### ISSUE TITLE
Additional run when using matrix

### ISSUE THREAD

**Issue by azzazzel** [nektos/act#507]
With 
```shell
❯ act --version
act version 0.2.19
```

and this workflow _(taken from [the docs](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#using-environment-variables-in-a-matrix) for the sole purpose of demonstrating the issue)_ :
```
name: Node.js CI
on: [push]
jobs:
  build:
    runs-on: ubuntu-latest
    strategy:
      matrix:
        include:
          - node-version: 10.x
            site: "prod"
            datacenter: "site-a"
          - node-version: 12.x
            site: "dev"
            datacenter: "site-b"
    steps:
      - name: Echo site details
        env:
          SITE: ${{ matrix.site }}
          DATACENTER: ${{ matrix.datacenter }}
        run: echo $SITE $DATACENTER
```

There are 3 runs instead of 2:

```
❯ act
[Node.js CI/build-1] 🚀  Start image=catthehacker/ubuntu:act-latest
[Node.js CI/build-3] 🧪  Matrix: map[datacenter:site-b node-version:12.x site:dev]
[Node.js CI/build-2] 🧪  Matrix: map[datacenter:site-a node-version:10.x site:prod]
[Node.js CI/build-3] 🚀  Start image=catthehacker/ubuntu:act-latest
[Node.js CI/build-2] 🚀  Start image=catthehacker/ubuntu:act-latest
[Node.js CI/build-2]   🐳  docker run image=catthehacker/ubuntu:act-latest entrypoint=["/usr/bin/tail" "-f" "/dev/null"] cmd=[]
[Node.js CI/build-3]   🐳  docker run image=catthehacker/ubuntu:act-latest entrypoint=["/usr/bin/tail" "-f" "/dev/null"] cmd=[]
[Node.js CI/build-1]   🐳  docker run image=catthehacker/ubuntu:act-latest entrypoint=["/usr/bin/tail" "-f" "/dev/null"] cmd=[]
[Node.js CI/build-3] ⭐  Run Echo site details
[Node.js CI/build-2] ⭐  Run Echo site details
[Node.js CI/build-1] ⭐  Run Echo site details
| prod site-a
| dev site-b
[Node.js CI/build-2]   ✅  Success - Echo site details
[Node.js CI/build-3]   ✅  Success - Echo site details
| 
[Node.js CI/build-1]   ✅  Success - Echo site details
```

As you can see one of them does not have the variables set up correctly. In the real scenario that causes jobs (and thus the whole workflow) to fail when running it with act.

**Comment by edtan** [comment]
This was fixed in #503.  Unfortunately, a new release with the fix hasn't been released yet, but if you build `act` from master, it should result in only 2 builds.

**Comment by azzazzel** [comment]
Thanks for the update. 
As I'm not dealing with Go on a daily basis I'd rather wait for the next release than investing time in installing and learning the toolchain needed to build Go apps. I hope I will not have to wait too long.
